k9mail/src/main/java/com/fsck/k9/activity/misc/ContactPictureLoader.java

```diff
@@ -271,4 +332,24 @@ public void cancel() {
         }
     }
 
+    private Uri getGravatarUri(String email) {
+        return Uri.parse("http://www.gravatar.com/avatar/" + getMD5String(email) + "?d=404");
+    }
+
+    private String getMD5String(String from) {
```

Comment:
we have [okio](http://square.github.io/okio/1.x/okio/) in our dependencies, there should be `ByteBuffer.md5()` available for this purpose